[feat] Add command-line option to manipulate the module path before executing any tests (#1606)
eth-cscs/reframe: docs/manpage.rst

   This option can also be set using the :envvar:`RFM_UNLOAD_MODULES` environment variable or the :js:attr:`unload_modules` general configuration parameter.

.. option:: --module-path=PATH

>    Use module path ``PATH`` before acting on any tests.

Reviewer: My problem with the `--module-path=/path/to/modules` is that it is a bit confusing. One could think that this would set the whole `MODULEPATH`. That's why I suggest to have the following three options:

1. `+/path/to/modules` -> `module use`
2. `-/path/to/modules` -> `module unuse`
3. `/path/to/modules` -> override completely `MODULEPATH`.

Since, as I suggest in the following, you need not provide a corresponding configuration parameter and/or environment variable, it should be quite straightforward to implement this logic.

   This option can also be set using the :envvar:`RFM_UNLOAD_MODULES` environment variable or the :js:attr:`unload_modules` general configuration parameter.

> .. option:: --unuse-module-path=PATH

Reviewer: Why not use a single option and use `+/-` to control addition and deletion of paths?

- `--module-path=+/path/to/add`
- `--module-path=-/path/to/remove`

Reviewer: If somebody passes it without `+/-` we could completely override the module path.

Author: I think that it's better to have only the `-` convension for unusing otherwise the module path should be used.

.. option:: --module-path=PATH

>    Use ``PATH`` to manipulate the ``MODULEPATH`` before acting on any tests.

Reviewer: ```suggestion
   Manipulate the ``MODULEPATH`` environment variable before acting on any tests.
```

>    Paths starting with the `-` character are going to be removed from the ``MODULEPATH``, while the ones starting with the `+` character are going to be added to it.

Reviewer: ```suggestion
   If ``PATH`` starts with the `-` character, it will be removed from the ``MODULEPATH``, whereas if it starts with the `+` character, it will be added to it.
```

.. option:: --module-path=PATH

   Manipulate the ``MODULEPATH`` environment variable before acting on any tests.

>    If ``PATH`` starts with the `-` character it will be removed from the ``MODULEPATH``, whereas if it starts with the `+` character, it will be added to it.

Reviewer: ```suggestion
   If ``PATH`` starts with the `-` character, it will be removed from the ``MODULEPATH``, whereas if it starts with the `+` character, it will be added to it.
```
